app/controllers/split_checkout_controller.rb

```diff
@@ -15,146 +15,55 @@ class SplitCheckoutController < ::BaseController
   helper OrderHelper
 
   def edit
-    return handle_redirect_from_stripe if valid_payment_intent_provided?
-
-    redirect_to_step unless checkout_step
+    redirect_to_step unless params[:step]
 
     OrderWorkflow.new(@order).next if @order.cart?
-
-    # This is only required because of spree_paypal_express. If we implement
-    # a version of paypal that uses this controller, and more specifically
-    # the #action_failed method, then we can remove this call
-    # OrderCheckoutRestart.new(@order).call
-    @order.errors.clear
-    @order.bill_address.errors.clear
-    @order.ship_address.errors.clear
-    @ship_address_same_as_billing = "1" if ship_address_matches_bill_address?
-  rescue Spree::Core::GatewayError => e
-    rescue_from_spree_gateway_error(e)
   end
 
   def update
-    load_shipping_method
-    handle_shipping_method_selection
-
-    confirm_or_update = confirm_order || update_order
-    if confirm_or_update && advance_order_state
+    if confirm_order || update_order
       clear_invalid_payments
       redirect_to_step
     else
-      if @shipping_method_id.blank?
-        @order.errors.add(:base, "no_shipping_method_selected")
-      end
-      flash.now[:error] = "#{I18n.t('split_checkout.errors.global')}"
+      flash.now[:error] = I18n.t('split_checkout.errors.global')
       render :edit
     end
   end
 
   private
 
-  def ship_address_matches_bill_address?
-    attrs_to_check = %w(firstname lastname address1 address2 state_id city zipcode phone country_id)
-    ((@order.bill_address.attributes.to_a -
-      @order.ship_address.attributes.to_a).map(&:first) & attrs_to_check).blank?
-  end
-
-  def handle_shipping_method_selection
-    return unless @shipping_method
-
-    populate_ship_address_params
-  end
-
-  def load_shipping_method
-    if params[:shipping_method_id]
-      @shipping_method = Spree::ShippingMethod.where(id: params[:shipping_method_id]).first
-      @shipping_method_id = params[:shipping_method_id]
-    else
-      @shipping_method = @order.shipping_method
-      @shipping_method_id = @shipping_method&.id
-    end
-  end
-
-  def populate_ship_address_params
-    return unless params.dig(:order, :ship_address_attributes).present? &&
-                  params.dig(:order, :bill_address_attributes).present?
-
-    if params.dig(:order, "Checkout.ship_address_same_as_billing") == "1"
-      params[:order][:ship_address_attributes] = params[:order][:bill_address_attributes]
-      return
-    end
-
-    address_attrs = [
-      :firstname,
-      :lastname,
-      :phone,
-      :address1,
-      :address2,
-      :city,
-      :state_id,
-      :zipcode,
-      :country_id,
-      :id
-    ]
-    address_attrs.each do |attr|
-      next if params[:order][:ship_address_attributes][attr].present?
-
-      params[:order][:ship_address_attributes][attr] = params[:order][:bill_address_attributes][attr]
-    end
```

Comment:
This was deleted but I don't think the problem it was solving is accounted for in any of the new code.

Reply:
We should start adding some tests soon to verify this stuff. What problem was it solving? We need to copy `:firstname`, `:lastname`, and `:phone` from `bill_address` to the `ship_address`. That's being done :ok_hand: